ignite/cmd/plugin.go

```diff
@@ -0,0 +1,253 @@
+package ignitecmd
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/ignite/cli/ignite/pkg/xgit"
+	"github.com/ignite/cli/ignite/services/plugin"
+)
+
+// plugins hold the list of plugin declared in the config.
+// A global variable is used so the list is accessible to the plugin commands.
+var plugins []*plugin.Plugin
+
+// loadPlugins tries to load all the plugins found in configuration.
+// If no configuration found, it returns w/o error.
+func loadPlugins(rootCmd *cobra.Command) error {
+	// NOTE(tb) Not sure if it's the right place to load this.
+	c, err := NewChainWithHomeFlags(rootCmd)
+	if err != nil {
+		// Binary is run outside of an chain app, plugins can't be loaded
+		return nil
+	}
+	plugins, err = plugin.Load(c)
+	if err != nil {
+		return err
+	}
+	// Link plugins to related commands
+	var loadErrors bool
+	for _, p := range plugins {
+		linkPluginCmds(rootCmd, p)
+		if p.Error != nil {
+			loadErrors = true
+		}
+	}
+	if loadErrors {
+		fmt.Println("Error(s) detected during plugin load:")
+		printPlugins()
+		os.Exit(1)
+	}
```

Comment:
@tbruyelle what do you think about just returning an error with the message `Error(s) detected during plugin load` instead of handling it and exiting inside the `loadPlugins` function.

This would allow the caller to check it and print the list of plugins. Plugins could be printed before the error message if the `loadPlugins` is called in a `PersistentPreRunE`, for example:

```go
PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
 // ...

 if err := loadPlugins(cmd); err != nil {
   printPlugins()

   return err
 }

 // ...
}
```